gobblin-data-management/src/main/java/gobblin/data/management/retention/profile/MultiCleanableDatasetFinder.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2015-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.retention.profile;
+
+import gobblin.config.client.ConfigClient;
+import gobblin.config.client.api.ConfigStoreFactoryDoesNotExistsException;
+import gobblin.config.client.api.VersionStabilityPolicy;
+import gobblin.config.store.api.ConfigStoreCreationException;
+import gobblin.config.store.api.VersionDoesNotExistException;
+import gobblin.data.management.dataset.Dataset;
+import gobblin.data.management.retention.DatasetCleaner;
+import gobblin.data.management.retention.dataset.finder.DatasetFinder;
+import gobblin.dataset.DatasetsFinder;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang.reflect.ConstructorUtils;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.collect.Lists;
+import com.typesafe.config.Config;
+
+
+/**
+ * A Clenable DatasetFinder that instantiates multiple DatasetFinders.
+ * <p>
+ * If {@link #DATASET_PROFILE_CLASS_KEY} is set, a single datasetFinder is created.
+ * Otherwise {@link #TAGS_TO_IMPORT_KEY} is used to find all the importedBy {@link URI}s from gobblin config management.
+ * The {@link Config} for each {@link URI} should have a {@link #DATASET_PROFILE_CLASS_KEY} set.
+ * </p>
+ *
+ */
+@Slf4j
+public class MultiCleanableDatasetFinder implements DatasetsFinder<Dataset> {
```

Comment:
Created a MultiDatasetFinder and subclass MultiCleanableDatasetFinder. The MultiDatasetFinder can be reused for MultiCopyableDatasetFinder. 
Also it is possible that a dataset has different finders for retention and distcp. I think it makes sense to not use a single key for both of them